src/main/scala/com/lucidworks/spark/example/ml/SVMExampleScala.scala

```diff
@@ -0,0 +1,161 @@
+        "soft_commit_secs" -> "10")
+      csvDF.write.format("solr").options(options).mode(SaveMode.Overwrite).save()
+    }
+
+    val contentFields = "tweet_txt"
+
+    var options = immutable.HashMap(
+      "zkhost" -> cli.getOptionValue("zkHost", DefaultZkHost),
+      "collection" -> cli.getOptionValue("collection", DefaultCollection),
+      "query" -> "+polarity:(0 OR 4) +tweet_txt:[* TO *]",
+      "fields" -> "id,polarity,tweet_txt",
+      "rows" -> "10000",
+      "splits" -> "true",
+      "split_field" -> "_version_",
+      "splits_per_shard" -> "8")
+
+    val sampleFraction = cli.getOptionValue("sample", "1.0").toDouble
+    var trainingDataFromSolr = sqlContext.read.format("solr").options(options).load()
+    trainingDataFromSolr = trainingDataFromSolr.sample(false, sampleFraction)
+
+    val inputCols = contentFields.split(" ").map(_.trim)
+
+    val stdTokLowerSchema = "{ \"analyzers\": [{ \"name\": \"std_tok_lower\", \"tokenizer\": { \"type\": \"standard\" },\n" +
+      "                \"filters\": [{ \"type\": \"lowercase\" }]}],\n" +
+      "  \"fields\": [{ \"regex\": \".+\", \"analyzer\": \"std_tok_lower\" }]}\n"
+
+    val numFeatures = cli.getOptionValue("numFeatures", DEFAULT_NUM_FEATURES).toInt
+    val numIterations = cli.getOptionValue("numIterations", DEFAULT_NUM_ITERATIONS).toInt
+
+    def RowtoLab(row: Row, numFeatures: Int, inputCols: Array[String], stdTokLowerSchema: String ): LabeledPoint = {
+      var textAnalyzer: LuceneTextAnalyzer = new LuceneTextAnalyzer(stdTokLowerSchema)
+      var hashingTF = new HashingTF(numFeatures)
+      var normalizer = new Normalizer()
+      val polarity = row.getString(row.fieldIndex("polarity"))
+      var fields = new java.util.HashMap[String, String]()
+      for(i <- 0 until inputCols.length){
+        val value = row.getString(row.fieldIndex(inputCols(i)))
+        if (value != null) {
+          fields.put(inputCols(i), value)
+        }
+      }
+      val analyzedFields = textAnalyzer.analyzeJava(fields)
+      var terms = new java.util.LinkedList[String]()
+      analyzedFields.values().asScala.toList.foreach(v => terms.addAll(v))
+      val sentimentLabel = if (("0" == polarity)) 0.toDouble else 1.toDouble
+      new LabeledPoint(sentimentLabel, normalizer.transform(hashingTF.transform(terms)))
+    }
+
+    var trainingData = trainingDataFromSolr.rdd.map(row => RowtoLab(row, numFeatures, inputCols, stdTokLowerSchema))
+    var standardScaler = new StandardScaler().fit(trainingData.map(x => x.features))
+    var trainRDD = trainingData.map(x => new LabeledPoint(x.label, standardScaler.transform(x.features)))
+    trainRDD = trainRDD.persist(StorageLevel.MEMORY_ONLY_SER)
+    val model = SVMWithSGD.train(trainRDD, numIterations)
+
+    options = immutable.HashMap("zkhost" -> cli.getOptionValue("zkHost", DefaultZkHost),
+      "collection" -> cli.getOptionValue("collection", DefaultCollection),
```

Comment:
Rather than re-using options (which can be confusing and difficult to debug), you should use new variable names